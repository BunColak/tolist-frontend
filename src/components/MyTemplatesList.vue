<template>
  <div class="sidebar">
    <h3 class="font-lg font-bold">My Templates</h3>
    <div class="mt-4 flex flex-col">
      <router-link
        v-for="template in templates"
        :key="template.id"
        :to="`/templates/${template.id}`"
        class="my-4"
        active-class="font-bold"
        >{{ template.title }}</router-link
      >
    </div>
  </div>
</template>

<script>
import Query from '../graphql/MyListTemplates.gql'

export default {
  name: 'MyTemplatesList',
  apollo: {
    templates: {
      query: Query,
      update: (data) => data.myListTemplates
    }
  }
}
</script>
