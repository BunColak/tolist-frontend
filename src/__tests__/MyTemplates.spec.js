import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import MyTemplatesList from '@/components/MyTemplatesList'

const mockTemplates = [
  {
    id: 1,
    title: 'Title 1'
  },
  {
    id: 2,
    title: 'Title 2'
  },
  {
    id: 3,
    title: 'Title 3'
  }
]

describe('MyTemplatesList', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(MyTemplatesList, {
      data () {
        return { templates: mockTemplates }
      },
      stubs: { RouterLink: RouterLinkStub }
    })
  })

  it('should render properly', async () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correct links', () => {
    wrapper = shallowMount(MyTemplatesList, {
      data () {
        return { templates: mockTemplates }
      },
      stubs: { RouterLink: RouterLinkStub }
    })
    expect(wrapper.findAll('a').length).toEqual(mockTemplates.length)
    wrapper.findAll('a').wrappers.forEach((wp, i) => {
      expect(wp.props('to')).toEqual(`/templates/${mockTemplates[i].id}`)
    })
  })
})
