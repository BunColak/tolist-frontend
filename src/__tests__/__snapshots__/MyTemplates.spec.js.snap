// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTemplatesList should render properly 1`] = `
<div class="sidebar">
  <h3 class="font-lg font-bold">My Templates</h3>
  <div class="mt-4 flex flex-col"><a class="my-4">Title 1</a><a class="my-4">Title 2</a><a class="my-4">Title 3</a></div>
</div>
`;
